tests/compute/vmi-configuration/vmi-definition.go

```diff
@@ -0,0 +1,1537 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2017-2023 Red Hat, Inc.
+ *
+ */
+
+package vmi_configuration
+
+import (
+	"context"
+	"fmt"
+	"runtime"
+	"strconv"
+	"strings"
+	"time"
+
+	"kubevirt.io/kubevirt/tests/compute"
+
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"kubevirt.io/kubevirt/pkg/virt-controller/services"
+
+	expect "github.com/google/goexpect"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+	k8sv1 "k8s.io/api/core/v1"
+	kubev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/utils/pointer"
+
+	v1 "kubevirt.io/api/core/v1"
+	virtv1 "kubevirt.io/api/core/v1"
+	"kubevirt.io/client-go/kubecli"
+	"kubevirt.io/client-go/log"
+
+	kvutil "kubevirt.io/kubevirt/pkg/util"
+	"kubevirt.io/kubevirt/tests"
+	"kubevirt.io/kubevirt/tests/decorators"
+	"kubevirt.io/kubevirt/tests/exec"
+	"kubevirt.io/kubevirt/tests/framework/kubevirt"
+	"kubevirt.io/kubevirt/tests/framework/matcher"
+	"kubevirt.io/kubevirt/tests/util"
+
+	"kubevirt.io/kubevirt/pkg/util/cluster"
+	"kubevirt.io/kubevirt/tests/console"
+	"kubevirt.io/kubevirt/tests/libnode"
+	"kubevirt.io/kubevirt/tests/libvmi"
+	"kubevirt.io/kubevirt/tests/libwait"
+	"kubevirt.io/kubevirt/tests/testsuite"
+	"kubevirt.io/kubevirt/tests/watcher"
+)
+
+var _ = compute.SIGDescribe("Configurations", func() {
```

Comment:
The idea behind `tests/compute` folder was to gradually move tests here. The primary prerequisites would be to make sure there are no duplicated tests and the tests are written in a consistent style (e.g using `libvmi`) or using best practices(e.g not ignoring warnings).

I found this commit/PR is not following this, need to admit self-proclaimed prerequisites. I don't see much of an improvement in "just moving files".

Could we work through each test one by one to ensure we improve our test suite?

Reply:
This PR is not about moving files around. My primary goal is to split up the tests to reduce the overall scope per file. This helps when running the tests inside the `check-tests-for-flakes` lane, so that lesser tests are being run overall.

I see this as an iterative process. Since the PR is already big enough, we should rather work on this iteratively, no?

Reply:
To explain further: the main drawback of big files is that [eventually the lane might not be run if the estimation finds too many tests would be executed](https://github.com/kubevirt/kubevirt/blob/d07f8d2b3a5d8d06b9fa2d65b4b2117685dc2936/automation/repeated_test.sh#L108).

Reply:
@xpivarc just to be clear: I am more than happy to work through this in a follow up task.

Reply:
Fair, could we just drop the move to `tests/compute` so we know these were not checked for duplicates and such?

Reply:
So IIUC we are missing a clear definition of the goals of such test refactoring PRs, right? Shoud we create a document for this?

Reply:
I think it would be fair to note that `tests/compute` should be subject to some requirements.

Reply:
> Fair, could we just drop the move to `tests/compute` so we know these were not checked for duplicates and such?

Since the series of commits starts with this, this would be a significant amount of work. Also I've been specifially asked to move them there.

Could we maybe just mark the files with a `FIXME: check for tests/compute` requirements?

Aside from that: what exactly are the requirements that should go into the README?